app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -124,6 +127,25 @@ def handle_remember_device_preference(remember_device_preference)
     save_remember_device_preference(remember_device_preference)
   end
 
+  def mfa_selection_attempt_count(auth_method)
+    user_session[:mfa_attempts] ||= {}
+    auth_method = auth_method.gsub('personal_key', 'personal-key')
+    attempt = { auth_method => 1 }
+    user_session[:mfa_attempts].merge!(attempt) { |_key, old_val, new_val| old_val + new_val }
```

Comment:
Feels like this could be simplified?

```suggestion
    user_session[:mfa_attempts] ||= {}
    user_session[:mfa_attempts][auth_method] ||= 0
    user_session[:mfa_attempts][auth_method] += 1
```

Reply:
Revised that to simplify.